src/util/event-to-promise.test.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-env jest */
+
+import 'core-js/fn/object/entries' // shim Object.entries
+import pull from 'lodash/pull'
+
+import eventToPromise from './event-to-promise'
+
+class MockEvent {
+    constructor() {
+        this.listeners = []
+    }
+
+    addListener(listener) {
+        this.listeners.push(listener)
+    }
+
+    removeListener(listener) {
+        pull(this.listeners, listener)
+    }
+
+    trigger(...args) {
+        this.listeners.forEach(listener => listener.apply(null, args))
+    }
+}
+
+describe('eventToPromise', () => {
+    test('should return a promise', () => {
+        const promise = eventToPromise({})
+        expect(promise).toBeInstanceOf(Promise)
+    })
+
+    test('should listen and unlisten to the events', async () => {
+        // We try both passing multiple events (for resolveOpts) and a single event (for rejectOpts).
+        const resolveOpts = [
+            {event: new MockEvent()},
+            {event: new MockEvent()},
+        ]
+        const rejectOpts = {event: new MockEvent()}
+        eventToPromise({
+            resolve: resolveOpts,
+            reject: rejectOpts,
+        })
+
+        // We use a bogus await statement to let any resolved promises invoke their callbacks.
+        // XXX Not sure if we can rely on this in every ES implementation.
+        await null
+        expect(resolveOpts[0].event.listeners.length).toBe(1)
+        expect(resolveOpts[1].event.listeners.length).toBe(1)
+        expect(rejectOpts.event.listeners.length).toBe(1)
+
+        // Trigger any of the events.
+        resolveOpts[1].event.trigger()
+
+        await null
+        expect(resolveOpts[0].event.listeners.length).toBe(0)
+        expect(resolveOpts[1].event.listeners.length).toBe(0)
+        expect(rejectOpts.event.listeners.length).toBe(0)
+    })
+
+    describe('should resolve with given value when a resolve-event occurs', () => {
```

Comment:
Just a design suggestion. It would be more consistent to use ```it``` instead of using ```describe```. Since in the original testing plan describe is used to denote collections of test belonging to modules. 
So we could have a structure like
->describe
--->test
----->it
Or we could interchange it and test. Whichever is more suitable. I notice now that it is not present in the official documentation, but the implementation works for sure.

Reply:
Your design sounds prettier indeed. I just followed the jest [documentation](http://facebook.github.io/jest/docs/en/api.html#describename-fn), which mentions nesting `describe` blocks but not nesting `test`s. I now see that `it` is documented too, it is shortly mentioned as being an alias for `test`, but that means it might not be nestable. Even if nesting tests would technically seem to work, I'd rather follow the documented approaches to avoid possible quirks later on (jest might change its implementation).